static/js/message_viewport.js

```diff
@@ -473,3 +476,10 @@ export function initialize() {
         stop_auto_scrolling();
     });
 }
+
+export function is_visible_and_focused() {
+    if (overlays.is_active() || !notifications.is_window_focused() || recent_topics.is_visible()) {
+        return false;
+    }
+    return true;
+}
```

Comment:
I was thinking it might be cleaner to replace the `recent_topics.is_visible` check with checking more directly whether the zfilt/zhome container is visible?

Reply:
Agreed, fixed.